### PR TITLE
SDI-129: ✨ Add in select absence sub reason

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: integration-tests/integration/activity/houseblockList.spec.js [hunk 1]
```diff
@@ -576,9 +578,11 @@ context('Houseblock list page list page', () => {
       expect(requestBody.offenderNo).to.eq('A1234AC')
       expect(requestBody.paid).to.eq(false)
       expect(requestBody.prisonId).to.eq('MDI')
+      expect(requestBody.absentReason).to.eq('RestDay')
+      expect(requestBody.absentSubReason).to.eq(undefined)
     })
 
-    houseblockPage.incentiveLevelCreated().should('be.visible')
+    houseblockPage.incentiveLevelCreated().should('not.exist')
```
Comment (petergphillips) [inline integration-tests/integration/activity/houseblockList.spec.js:585]: switched to non IEP reason - `RestDay` hence iep dialog not now shown

File: src/Attendance/AttendanceOtherForm/AttendanceOtherForm.js [hunk 2]
```diff
@@ -122,6 +129,10 @@ export function AttendanceOtherForm({
     if (!pay) return []
     return pay === 'yes' ? absentReasons.paidReasons : absentReasons.unpaidReasons
   }
+  const getAbsentSubReasons = (pay, reason) => {
+    if (!pay || !shouldTriggerSubReason(reason)) return []
```
Comment (petergphillips) [inline src/Attendance/AttendanceOtherForm/AttendanceOtherForm.js:133]: sub reason only displayed if certain absent reasons selected

File: src/Attendance/AttendanceOtherForm/AttendanceOtherForm.js [hunk 3]
```diff
@@ -143,7 +155,12 @@ export function AttendanceOtherForm({
       render={({ handleSubmit, submitting, pristine, submitError: errors, values }) => (
         <form onSubmit={handleSubmit}>
           {errors && (
-            <ErrorSummary onHandleErrorClick={onHandleErrorClick} heading="There is a problem" errors={errors} />
+            <ErrorSummary
```
Comment (andymarke) [inline src/Attendance/AttendanceOtherForm/AttendanceOtherForm.js:158]: I never knew there was still React in DPS

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**